File: app/src/main/java/com/letianpai/robot/desktop/callback/AppInstallSuccessCallback.java
package com.letianpai.robot.desktop.callback;

/**
 * 应用状态更新回调
 */

public class AppInstallSuccessCallback {

    private AppStoreInstallResultListener mAppStatusUpdateListener;

    private static class AppStatusUpdateCallbackHolder {
        private static AppInstallSuccessCallback instance = new AppInstallSuccessCallback();
    }

    private AppInstallSuccessCallback() {

    }

    public static AppInstallSuccessCallback getInstance() {
        return AppStatusUpdateCallbackHolder.instance;
    }

    public interface AppStoreInstallResultListener {
        void appInstallSuccess(String packageName,String appName);

    }

    public void setAppStatusUpdateListener(AppStoreInstallResultListener listener) {
        this.mAppStatusUpdateListener = listener;
    }

    public void responseAppInstallSuccess(String packageName,String appName) {
        if (mAppStatusUpdateListener != null) {
            mAppStatusUpdateListener.appInstallSuccess(packageName,appName);
        }
    }

}

File: app/src/main/java/com/letianpai/robot/desktop/callback/AppInstallSuccessCallback.kt
package com.letianpai.robot.desktop.callback

/**
 * Application status update callbacks
 */
class AppInstallSuccessCallback private constructor() {
    private var mAppStatusUpdateListener: AppStoreInstallResultListener? = null

    private object AppStatusUpdateCallbackHolder {
        val instance: AppInstallSuccessCallback = AppInstallSuccessCallback()
    }

    fun interface AppStoreInstallResultListener {
        fun appInstallSuccess(packageName: String?, appName: String?)
    }

    fun setAppStatusUpdateListener(listener: AppStoreInstallResultListener?) {
        this.mAppStatusUpdateListener = listener
    }

    fun responseAppInstallSuccess(packageName: String?, appName: String?) {
        if (mAppStatusUpdateListener != null) {
            mAppStatusUpdateListener!!.appInstallSuccess(packageName, appName)
        }
    }

    companion object {
        val instance: AppInstallSuccessCallback
            get() = AppStatusUpdateCallbackHolder.instance
    }
}
